projects/dotcms-field-elements/src/components/dot-date/dot-date.e2e.ts

```diff
@@ -1,131 +1,385 @@
+            });
+        });
+    });
+
+    describe('@Events', () => {
+        let spyStatusChangeEvent: EventSpy;
+        let spyValueChangeEvent: EventSpy;
+
+        beforeEach(async () => {
+            spyStatusChangeEvent = await page.spyOnEvent('statusChange');
+            spyValueChangeEvent = await page.spyOnEvent('valueChange');
+        });
+
+        describe('value and status changes', () => {
+            it('should send value when dot-input-calendar send it', async () => {
+                inputCalendar.triggerEvent('_valueChange', {
+                    detail: {
+                        name: '',
+                        value: '2019-01-01'
+                    }
+                });
+                await page.waitForChanges();
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: '',
+                    value: '2019-01-01'
+                });
+            });
+
+            it('should emit status and value on Reset', async () => {
+                await element.callMethod('reset');
+                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                    name: '',
+                    status: {
+                        dotPristine: true,
+                        dotTouched: false,
+                        dotValid: true
+                    }
+                });
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: '',
+                    value: ''
+                });
+            });
+
+            it('should send status and value change and stop dot-input-calendar events', async () => {
+                const evt_statusChange = await page.spyOnEvent('_statusChange');
+                const evt_valueChange = await page.spyOnEvent('_valueChange');
+
+                inputCalendar.triggerEvent('_valueChange', {
+                    detail: {
+                        name: '',
+                        value: '2019-01-01'
+                    }
+                });
+                inputCalendar.triggerEvent('_statusChange', {
+                    detail: {
+                        name: '',
+                        status: {
+                            dotPristine: false,
+                            dotTouched: true,
+                            dotValid: true
+                        }
+                    }
+                });
+                await page.waitForChanges();
+                expect(spyStatusChangeEvent).toHaveReceivedEventDetail({
+                    name: '',
+                    status: {
+                        dotPristine: false,
+                        dotTouched: true,
+                        dotValid: true
+                    }
+                });
+                expect(spyValueChangeEvent).toHaveReceivedEventDetail({
+                    name: '',
+                    value: '2019-01-01'
+                });
+                expect(evt_statusChange.events).toEqual([]);
```

Comment:
What are you testing here?

Reply:
that the `stopImmediatePropagation` for `evt_statusChange` happened.